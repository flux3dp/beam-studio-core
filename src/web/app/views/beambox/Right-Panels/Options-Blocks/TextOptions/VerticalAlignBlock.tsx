import React from 'react';

import i18n from 'helpers/i18n';
import ObjectPanelItem from 'app/views/beambox/Right-Panels/ObjectPanelItem';
import { useIsMobile } from 'helpers/system-helper';
import { VerticalAlign } from 'app/actions/beambox/textPathEdit';

interface Props {
  value: VerticalAlign;
  onValueChange: (val: VerticalAlign) => void;
}

export default function VerticalAlignBlock({ value, onValueChange }: Props): JSX.Element {
  const LANG = i18n.lang.beambox.right_panel.object_panel;
  const label = LANG.option_panel.vertical_align;
  const isMobile = useIsMobile();
  if (isMobile) {
    const options = [
      { value: VerticalAlign.BOTTOM, label: LANG.bottom_align },
      { value: VerticalAlign.MIDDLE, label: LANG.middle_align },
      { value: VerticalAlign.TOP, label: LANG.top_align },
    ];
    return (
      <ObjectPanelItem.Select
        id="vertical_align"
        selected={options.find((option) => option.value === value)}
        options={options}
        onChange={onValueChange}
        label={label}
      />
    );
  }
  const options = [
    <option key={VerticalAlign.BOTTOM} value={VerticalAlign.BOTTOM}>{LANG.bottom_align}</option>,
    <option key={VerticalAlign.MIDDLE} value={VerticalAlign.MIDDLE}>{LANG.middle_align}</option>,
    <option key={VerticalAlign.TOP} value={VerticalAlign.TOP}>{LANG.top_align}</option>,
  ];
  return (
    <div className="option-block">
      <div className="label">{label}</div>
      <div className="select-container">
        <select
          value={value}
          onChange={(e) => onValueChange(Number(e.target.value))}
        >
          {options}
        </select>
      </div>
    </div>
  );
}
