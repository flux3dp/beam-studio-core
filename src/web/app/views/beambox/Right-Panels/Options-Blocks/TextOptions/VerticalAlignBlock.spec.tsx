/* eslint-disable import/first */
import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

enum VerticalAlign {
  BOTTOM = 0,
  MIDDLE = 1,
  TOP = 2,
}

jest.mock('app/actions/beambox/textPathEdit', () => ({
  VerticalAlign,
}));

const isMobile = jest.fn();
jest.mock('helpers/system-helper', () => ({
  isMobile: () => isMobile(),
}));

jest.mock('helpers/i18n', () => ({
  lang: {
    beambox: {
      right_panel: {
        object_panel: {
          option_panel: {
            vertical_align: 'Vertical Align',
          },
          bottom_align: 'Bottom Align',
          middle_align: 'Middle Align',
          top_align: 'Top Align',
        },
      },
    },
  },
}));

import VerticalAlignBlock from './VerticalAlignBlock';

describe('test VerticalAlignBlock', () => {
  test('should render correctly', () => {
    const onValueChange = jest.fn();
    const wrapper = shallow(
      <VerticalAlignBlock value={VerticalAlign.BOTTOM} onValueChange={onValueChange} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('div.option-block').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();

    wrapper.find('select').simulate('change', { target: { value: VerticalAlign.TOP } });
    expect(onValueChange).toHaveBeenCalledTimes(1);
    expect(onValueChange).toHaveBeenNthCalledWith(1, VerticalAlign.TOP);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should render correctly in mobile', () => {
    isMobile.mockReturnValue(true);
    const onValueChange = jest.fn();
    const wrapper = shallow(
      <VerticalAlignBlock value={VerticalAlign.BOTTOM} onValueChange={onValueChange} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
