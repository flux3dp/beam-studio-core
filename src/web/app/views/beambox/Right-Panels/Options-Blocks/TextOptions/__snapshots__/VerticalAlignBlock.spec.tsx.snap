// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="option-block"
>
  <div
    className="label"
  >
    Vertical Align
  </div>
  <div
    className="select-container"
  >
    <select
      onChange={[Function]}
      value={0}
    >
      <option
        key="0"
        value={0}
      />
      <option
        key="1"
        value={1}
      />
      <option
        key="2"
        value={2}
      />
    </select>
  </div>
</div>
`;

exports[`should render correctly 2`] = `
<div
  className="option-block"
>
  <div
    className="label"
  >
    Vertical Align
  </div>
  <div
    className="select-container"
  >
    <select
      onChange={[Function]}
      value={0}
    >
      <option
        key="0"
        value={0}
      />
      <option
        key="1"
        value={1}
      />
      <option
        key="2"
        value={2}
      />
    </select>
  </div>
</div>
`;

exports[`should render correctly 3`] = `
<div
  className="option-block"
>
  <div
    className="label"
  >
    Vertical Align
  </div>
  <div
    className="select-container"
  >
    <select
      onChange={[Function]}
      value={0}
    >
      <option
        key="0"
        value={0}
      />
      <option
        key="1"
        value={1}
      />
      <option
        key="2"
        value={2}
      />
    </select>
  </div>
</div>
`;
